packages/components/Menu/src/Menu/Menu.tsx

```diff
@@ -28,7 +28,7 @@ export const Menu = stagedComponent((props: MenuProps) => {
     return (
       <MenuProvider value={contextValue}>
         {menuTrigger}
-        {state.open && menuPopover}
+        {state.open && state.shouldFocusOnContainer !== undefined && menuPopover}
```

Comment:
Nit: can we add a comment, as I feel like this is otherwise not clear?

Reply:
I added a comment, does it help?